APMrover2/mode_guided.cpp

```diff
@@ -48,7 +52,11 @@ void ModeGuided::update()
                 calc_steering_to_heading(_desired_yaw_cd);
                 calc_throttle(calc_reduced_speed_for_turn_or_distance(_desired_speed), true, true);
             } else {
-                stop_vehicle();
+                if (rover.is_boat()) {
+                    start_loiter();
```

Comment:
What happens if `start_loiter` fails here?  What state does that leave us in?

Reply:
start_loiter should probably return a bool to notify whether it could get into loiter or not.

Reply:
@peterbarker It should be stopped when fail to get in Loiter. I fixed it.
@rmackay9 Thank you for your advice.